flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/FlinkHints.java

```diff
@@ -75,4 +90,88 @@ public static Map<String, String> mergeTableOptions(
         newProps.putAll(hints);
         return Collections.unmodifiableMap(newProps);
     }
+
+    public static Optional<String> getTableAlias(RelNode node) {
+        if (node instanceof Hintable) {
+            Hintable aliasNode = (Hintable) node;
+            List<String> aliasNames =
+                    aliasNode.getHints().stream()
+                            .filter(h -> h.hintName.equalsIgnoreCase(FlinkHints.HINT_ALIAS))
+                            .flatMap(h -> h.listOptions.stream())
+                            .collect(Collectors.toList());
+            if (aliasNames.size() > 0) {
+                return Optional.of(aliasNames.get(0));
+            } else {
+                if (canTransposeToTableScan(node)) {
+                    return getTableAlias(node.getInput(0));
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static boolean canTransposeToTableScan(RelNode node) {
+        // TODO support look up join
+        return node instanceof LogicalProject // computed column on table
+                || node instanceof LogicalFilter;
+    }
+
+    /** Returns the qualified name of a table scan, otherwise returns empty. */
+    public static Optional<String> getTableName(RelOptTable table) {
+        if (table == null) {
+            return Optional.empty();
+        }
+
+        String tableName;
+        if (table instanceof TableSourceTable) {
```

Comment:
we can use `FlinkPreparingTableBase#getNames` to get table name safely since there're two many sub-classes